projects/client/RabbitMQ.Client/src/util/NetworkBinaryReader.cs

```diff
@@ -53,147 +54,207 @@ namespace RabbitMQ.Util
     /// Relies on BinaryReader always being little-endian.
     /// </para>
     /// </remarks>
-    public class NetworkBinaryReader : BinaryReader
+    public class NetworkBinaryReader 
     {
-        // Not particularly efficient. To be more efficient, we could
```

Comment:
I think it were good to keep an explanation similar to this comment, that explains why BinaryReader is not used here as a basis.